src/zope/app/form/browser/widgets.rst

```diff
@@ -13,56 +13,60 @@ description
 Simple stuff.
 
 Our Poll product holds an editable list of the `PollOption` instances.
-This is shown in the ``poll.py`` source below::
 
-    from persistent import Persistent
-    from interfaces import IPoll, IPollOption
-    from zope.interface import implements, classImplements
+First the Schemas are defined in the ``interfaces.py`` file below::
+
+    >>> from zope.interface import Interface
+    >>> from zope.schema import Object, Tuple, TextLine
+    >>> from zope.schema.interfaces import ITextLine
+    >>> from zope.i18nmessageid import MessageFactory
+
+    >>> _ = MessageFactory("poll")
+
+    >>> class IPollOption(Interface):
+    ...    label = TextLine(title=u'Label', min_length=1)
+    ...    description = TextLine(title=u'Description', min_length=1)
+
+    >>> class IPoll(Interface):
+    ...    options = Tuple(title=u'Options',
+    ...        value_type=Object(IPollOption, title=u'Poll Option'))
+    ...
+    ...    def getResponse(option): "get the response for an option"
+    ...
+    ...    def choose(option): 'user chooses an option'
+
+The implementation is shown in the ``poll.py`` source below::
+
+    >>> from persistent import Persistent
+    >>> from zope.interface import implementer, classImplements
+
+    >>> @implementer(IPollOption)
+    ... class PollOption(Persistent, object):
+    ...       pass
+
+    >>> @implementer(IPoll)
+    ... class Poll(Persistent, object):
+    ...
+    ...
+    ...  def getResponse(self, option):
```

Comment:
There's some out-of-the-box support (set softtabstop=4 and hitting <tab>/<backspace> indents/dedents the rest of the line to the right mutliple of four) and some plugins (to get Python syntax higlighting in doctests in ReStructuredText files).  I also like visual block mode for adding the doctest prefix to a bunch of lines in one go, when you move existing code into a doctest.

IIRC I once tried very hard to make autoindentation work correctly (i.e. insert a newline, four spaces, three dots, a space, and then the right number of spaces for Python code), but gave up without achieving success.